### PR TITLE
SITL: Document Airspeed Params

### PR DESCRIPTION
**Pull request by antholuo**
Unsure how to document ARSPD2 Params

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -2150,19 +2150,14 @@ def get_sim_parameter_documentation_get_whitelist(self):
             "SIM_ADSB_COUNT",
             "SIM_ADSB_RADIUS",
             "SIM_ADSB_TX",
-            "SIM_ARSPD2_FAIL",
             "SIM_ARSPD2_FAILP",
```
Comment (antholuo) [inline Tools/autotest/common.py:2153]: For ex, ARSPD2_FAIL exists in common.py but doesn't explicitly exist in SITL.cpp

File: libraries/SITL/SITL.cpp [hunk 2]
```diff
@@ -475,9 +475,28 @@ const AP_Param::GroupInfo SIM::AirspeedParm::var_info[] = {
     // @Values: 0:Disabled, 1:Enabled
     // @User: Advanced
     AP_GROUPINFO("FAIL",    3, SIM::AirspeedParm,  fail, 0),
+    // @Param: ARSPD_FAILP
```
Comment (IamPete1) [inline libraries/SITL/SITL.cpp:478]: The prefix is in the wrong spot, This should just be FAILP. The one level up where these are included you add the prefix. Like gripper for example. 

https://github.com/ArduPilot/ardupilot/blob/9b0639557338792356f50e39046e216d215be32b/libraries/SITL/SITL.cpp#L217-L219

Then the documentation should work for both airspeed 1 and 2.

File: libraries/SITL/SITL.cpp [hunk 3]
```diff
@@ -469,15 +478,34 @@ const AP_Param::GroupInfo SIM::AirspeedParm::var_info[] = {
...
     AP_GROUPINFO("RND",     1, SIM::AirspeedParm,  noise, 2.0),
     AP_GROUPINFO("OFS",     2, SIM::AirspeedParm,  offset, 2013),
-    // @Param: ARSPD_FAIL
+    // @Param: FAIL
     // @DisplayName: Airspeed sensor failure
     // @Description: Simulates Airspeed sensor 1 failure
     // @Values: 0:Disabled, 1:Enabled
     // @User: Advanced
     AP_GROUPINFO("FAIL",    3, SIM::AirspeedParm,  fail, 0),
+    // @Param: FALIP
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:487]:
```suggestion
    // @Param: FAILP
```

File: libraries/SITL/SITL.cpp [hunk 4]
```diff
@@ -432,8 +439,10 @@ const AP_Param::GroupInfo SIM::var_info3[] = {
     // @User: Advanced
     AP_GROUPINFO("UART_LOSS", 42, SIM,  uart_byte_loss_pct, 0),
 
+    // @Group: ARPSD_
```
Comment (peterbarker) [inline libraries/SITL/SITL.cpp:442]:
```suggestion
    // @Group: ARSPD_
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

**Review by peterbarker — CHANGES_REQUESTED** [review]
OK, so the parser isn't coping with the setup for these parameters, and I don't think it will be easy to make it do so.

It believes that anything that looks like a parameter or group in this file is at the same level.  So here we get `SIM_PITOT`, `SIM_FAILP` etc etc.  The `ARSPD` group parameter declaration comment is missing `@Path`, and we can't specify "this file" in there as that's infinite recursion.  It's the @Path and recursion thing which would tack on `ARSPD_` to the parameter name base to get `SIM_ARSPD_FAILP`.

Easiest fix is to move the definition of `const AP_Param::GroupInfo SIM::AirspeedParm::var_info[] = {` into a SITL_airspeed.cpp into `AP_HAL_SITL/sitl_airspeed.cpp` - and I don't think that's a terrible thing.  Then just add `@Path` to point to that file.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**